src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -898,13 +898,6 @@ public static bool IsAwaitable(this ISymbol symbol, SemanticModel semanticModel,
                 {
                     return false;
                 }
-
-                // dynamic
-                if (methodSymbol.ReturnType.TypeKind == TypeKind.Dynamic &&
```

Comment:
Should we rename the method to IsAwaitableNonDynamic or something? Technically `dynamic` is awaitable. I worry somebody else is going to undo this at some point because it's "wrong".